tests/test_ilastik/widgets/test_colorDialog.py

```diff
@@ -0,0 +1,127 @@
+    brush_color, pmap_color = default_colors
+    dlg = ColorDialog(brush_color, pmap_color)
+    qtbot.addWidget(dlg)
+    qtbot.waitExposed(dlg)
+
+    return dlg
+
+
+@pytest.fixture
+def color_dialog_nonmatching(qtbot, nonmatching_colors: ColorTuple) -> ColorDialog:
+    brush_color, pmap_color = nonmatching_colors
+    dlg = ColorDialog(brush_color, pmap_color)
+    qtbot.addWidget(dlg)
+    qtbot.waitExposed(dlg)
+
+    return dlg
+
+
+def test_construct(color_dialog: ColorDialog, default_colors: ColorTuple):
+    assert color_dialog.brushColor() == default_colors.brush_color
+    assert color_dialog.pmapColor() == default_colors.pmap_color
+    assert color_dialog.brushColor() == color_dialog.pmapColor()
+    assert color_dialog.ui.linkColorCheckBox.checkState()
+
+
+@patch("ilastik.widgets.labelListView.ColorDialog._getColor", lambda *x: QColor("yellow"))
+def test_sync_brush_color(color_dialog: ColorDialog, default_colors: ColorTuple):
+    color_dialog.onBrushColor()
+    assert color_dialog.brushColor() == QColor("yellow")
+    assert color_dialog.pmapColor() == QColor("yellow")
+
+
+@patch("ilastik.widgets.labelListView.ColorDialog._getColor", lambda *x: QColor("rose"))
+def test_sync_pmap_color(color_dialog: ColorDialog, default_colors: ColorTuple):
+    color_dialog.onPmapColor()
+    assert color_dialog.brushColor() == QColor("rose")
+    assert color_dialog.pmapColor() == QColor("rose")
+
+
+def test_construct_nonmatching(color_dialog_nonmatching: ColorDialog, nonmatching_colors: ColorTuple):
+    assert color_dialog_nonmatching.brushColor() == nonmatching_colors.brush_color
+    assert color_dialog_nonmatching.pmapColor() == nonmatching_colors.pmap_color
+    assert not color_dialog_nonmatching.ui.linkColorCheckBox.checkState()
+
+
+@patch("ilastik.widgets.labelListView.ColorDialog._getColor", lambda *x: QColor("yellow"))
+def test_sync_off_brush_color(color_dialog_nonmatching: ColorDialog, nonmatching_colors: ColorTuple):
+    color_dialog_nonmatching.onBrushColor()
+    assert color_dialog_nonmatching.brushColor() == QColor("yellow")
+    assert color_dialog_nonmatching.pmapColor() == nonmatching_colors.pmap_color
+
+
+@patch("ilastik.widgets.labelListView.ColorDialog._getColor", lambda *x: QColor("rose"))
+def test_sync_off_pmap_color(color_dialog_nonmatching: ColorDialog, nonmatching_colors: ColorTuple):
+    color_dialog_nonmatching.onPmapColor()
+    assert color_dialog_nonmatching.brushColor() == nonmatching_colors.brush_color
+    assert color_dialog_nonmatching.pmapColor() == QColor("rose")
+
+
+def test_sync_toggle(color_dialog: ColorDialog, default_colors: ColorTuple):
+    assert color_dialog.brushColor() == default_colors.brush_color
+    assert color_dialog.pmapColor() == default_colors.pmap_color
+
+    color_dialog.ui.linkColorCheckBox.toggle()
+    assert color_dialog.brushColor() == default_colors.brush_color
+    assert color_dialog.pmapColor() == default_colors.brush_color
```

Comment:
```suggestion
@patch("ilastik.widgets.labelListView.ColorDialog._getColor", lambda *x: QColor("blue"))
def test_sync_toggle_to_desync(color_dialog: ColorDialog, default_colors: ColorTuple):
    assert color_dialog.brushColor() == default_colors.brush_color
    assert color_dialog.pmapColor() == default_colors.pmap_color

    color_dialog.ui.linkColorCheckBox.toggle()
    assert color_dialog.brushColor() == color_dialog.pmapColor() == default_colors.brush_color

    color_dialog.onBrushColor()
    assert color_dialog.brushColor() == QColor("blue")
    assert color_dialog.pmapColor() == default_colors.brush_color

def test_sync_toggle_to_sync(color_dialog_nonmatching: ColorDialog, nonmatching_colors: ColorTuple):
    assert color_dialog_nonmatching.brushColor() == nonmatching_colors.brush_color
    assert color_dialog_nonmatching.pmapColor() == nonmatching_colors.pmap_color

    color_dialog_nonmatching.ui.linkColorCheckBox.toggle()
    assert color_dialog_nonmatching.brushColor() == nonmatching_colors.brush_color
    assert color_dialog_nonmatching.pmapColor() == nonmatching_colors.brush_color
```

These tests are fun to play around with 😅